jax/_src/api.py

```diff
@@ -1788,6 +1788,10 @@ def pmap(
     :py:func:`pmap` compiles ``fun``, so while it can be combined with
     :py:func:`jit`, it's usually unnecessary.
 
+  :py:func:`pmap` requires that all of the participating devices are identical.
```

Comment:
Is identical a very strong word here? (it can means that its the same device too)

I know you example clarifies it, but can we improve the first statement?

Reply:
I can't think of a better word, so LGTM anyways

Reply:
I do mean "identical". The GPUs must be exactly alike in every way.

Added a note that the same device cannot participate twice.